src/ansys/aedt/core/visualization/post/common.py

```diff
@@ -0,0 +1,2417 @@
+        """Logger."""
+        return self._app.logger
+
+    @property
+    def _desktop(self):
+        """Desktop."""
+        return self._app._desktop
+
+    @property
+    def _odesign(self):
+        """Design."""
+        return self._app._odesign
+
+    @property
+    def _oproject(self):
+        """Project."""
+        return self._app._oproject
+
+    @property
+    def modeler(self):
+        """Modeler."""
+        return self._app.modeler
+
+    @property
+    def post_solution_type(self):
+        """Design solution type.
+
+        Returns
+        -------
+        type
+            Design solution type.
+        """
+        return self._app.solution_type
+
+    @property
+    def all_report_names(self):
+        """List of all report names.
+
+        Returns
+        -------
+        list
+
+        References
+        ----------
+
+        >>> oModule.GetAllReportNames()
+        """
+        return list(self.oreportsetup.GetAllReportNames())
+
+    @pyaedt_function_handler(PlotName="plot_name")
+    def copy_report_data(self, plot_name, paste=True):
+        """Copy report data as static data.
+
+        Parameters
+        ----------
+        plot_name : str
+            Name of the report.
+        paste : bool, optional
+            Whether to paste the report. The default is ``True``.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oModule.CopyReportsData
+        >>> oModule.PasteReports
+        """
+        self.oreportsetup.CopyReportsData([plot_name])
+        if paste:
+            self.paste_report_data()
+        return True
+
+    @pyaedt_function_handler()
+    def paste_report_data(self):
+        """Paste report data as static data.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+        >>> oModule.PasteReports
+        """
+        self.oreportsetup.PasteReports()
+        return True
+
+    @pyaedt_function_handler()
+    def delete_report(self, plot_name=None):
+        """Delete all reports or specific report.
+
+        Parameters
+        ----------
+        plot_name : str, optional
+            Name of the plot to delete. The default  value is ``None`` and in this case, all reports are deleted.
+
+        Returns
+        -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        References
+        ----------
+
+        >>> oModule.DeleteReports
+        """
+        try:
+            if plot_name:
+                self.oreportsetup.DeleteReports([plot_name])
+                for plot in self.plots:
+                    if plot.plot_name == plot_name:
+                        self.plots.remove(plot)
+            else:
+                self.oreportsetup.DeleteAllReports()
+                if is_ironpython:  # pragma: no cover
```

Comment:
We can keep that and update it later on (while removing everything related to ironpython code), or you can chose to remove it now. I'll probably not catch every changes related to ironpython in that PR.